### PR TITLE
Add ability for which builtin to process multiple arguments

### PR DESCRIPTION
**Pull request by Sag0Sag0**

### CODE REVIEW

File: src/builtins/mod.rs [hunk 1]
```diff
@@ -585,34 +585,29 @@ fn builtin_which(args: &[&str], shell: &mut Shell) -> i32 {
...
-        println!("{}: function", command);
-        SUCCESS
-    } else {
-        for path in env::var("PATH").unwrap_or("/bin".to_string()).split(sys::PATH_SEPARATOR) {
-            let executable = Path::new(path).join(command);
-            if executable.is_file() {
-                println!("{}", executable.display());
-                return SUCCESS;
+    let mut result = SUCCESS;
+    'outer: for i in 1..args.len() {
```
Comment (mmstick) [inline src/builtins/mod.rs]: Would be more efficient to write this as:

```rust
`outer: for arg in &args[1..] { }
```

File: src/builtins/mod.rs [hunk 2]
```diff
@@ -585,34 +585,29 @@ fn builtin_which(args: &[&str], shell: &mut Shell) -> i32 {
...
+        let command = args[i];
+        if let Some(alias) = shell.variables.aliases.get(command) {
+            println!("{}: alias to {}", command, alias);
+            continue;
+        } else if shell.builtins.contains_key(command) {
+            println!("{}: built-in shell command", command);
+            continue;
+        } else if shell.functions.contains_key(command) {
```
Comment (mmstick) [inline src/builtins/mod.rs]: Should functions come before builtins?

Comment (Sag0Sag0) [reply src/builtins/mod.rs]: That was how it was before I edited it.

File: src/builtins/mod.rs (hunk 1 above)
Comment (Sag0Sag0) [reply src/builtins/mod.rs]: Yes, however that causes an error.

File: src/builtins/mod.rs (hunk 2 above)
Comment (mmstick) [reply src/builtins/mod.rs]: May want to change it while you're at it.

Comment (Sag0Sag0) [reply src/builtins/mod.rs]: Sure.

File: src/builtins/mod.rs (hunk 1 above)
Comment (Sag0Sag0) [reply src/builtins/mod.rs]: Just worked out the error was being caused by something else. Implementing it now.

File: src/builtins/mod.rs (hunk 2 above)
Comment (mmstick) [reply src/builtins/mod.rs]: I believe you could also just write this as so:

```rust
'outer: for &command in &args[1..] { }
```

And then not have to manually dereference it,

Comment (Sag0Sag0) [reply src/builtins/mod.rs]: Thanks, sorry about the mistakes, I only recently began learning rust.

Comment (mmstick) [reply src/builtins/mod.rs]: It's fine. This is part of the purpose of submitting and reviewing pull requests -- pointing out tricks that you may not be aware of yet.

**Comment by mmstick** [comment]
Thanks!

**Comment by logankaser** [comment]
Does this mean read can have a prompt now? That would be very nice.

**Pull request merged**